### PR TITLE
SALTO-5324: Remove serviceDeskId

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I removed the use of serviceDeskID and change to projectId identifier. 

---

_Additional context for reviewer_
In jira, a project has many unique Identifiers - ProjectKey, projectId, serviceDeskId.
Currently I see that we don't need to use serviceDeskId and we can use different Identifiers as shown [here](https://developer.atlassian.com/cloud/jira/service-desk/rest/intro/#project-identifiers).
In order to reduce api calles and re use common code - I changed the identifier from serviceDeskId to projectId

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

**Review by omrilit — APPROVED** [review]

File: packages/jira-adapter/src/adapter.ts [hunk 1]
```diff
@@ -572,23 +574,28 @@ export default class JiraAdapter implements AdapterOperations {
...
       .filter(isInstanceElement)
       .filter(project => project.value.projectTypeKey === SERVICE_DESK)
-      .filter(project => {
-        if (project.value.serviceDeskId?.id === undefined) {
-          log.debug(`Service desk project ${project.elemID.name} does not have a service desk id`)
-          return false
-        }
-        return true
-      })
+      .filter(project => serviceDeskProjectIds.includes(project.value.id)))
```
Comment (omrilit) [inline packages/jira-adapter/src/adapter.ts:594]: Worth adding a log indicating why the project was filtered so we could pinpoint the problem if needed, WDYT?

File: packages/jira-adapter/src/change_validators/jsm/jsm_permissions.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      paginator(paginationArgs, page => makeArray(page.values) as clientUtils.ResponseValue[])
+    )).flat().map(project => project.projectId)
+
+    return awu(changes)
+      .filter(isInstanceChange)
+      .map(getChangeData)
+      .filter(instance => SUPPORTED_TYPES.has(instance.elemID.typeName))
+      .filter(instance => {
+        try {
+          const projectParent = getParent(instance)
```
Comment (omrilit) [inline packages/jira-adapter/src/change_validators/jsm/jsm_permissions.ts:60]: `getParent` could throw an exception that will be caught here and the error message will not be relevant to it

File: packages/jira-adapter/src/change_validators/jsm/jsm_permissions.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          return !serviceDeskProjectIds.includes(projectParent.value.id)
+        } catch (e) {
+          return false
+        }
+      })
+      .map(instance => ({
+        elemID: instance.elemID,
+        severity: 'Error' as SeverityLevel,
+        message: 'Cannot deploy a JSM type without JSM permissions',
+        detailedMessage: `Cannot deploy ${instance.elemID.name} without JSM permissions`,
```
Comment (omrilit) [inline packages/jira-adapter/src/change_validators/jsm/jsm_permissions.ts:70]: Should Lior go over this one?

Comment (idoamit8) [reply packages/jira-adapter/src/change_validators/jsm/jsm_permissions.ts:70]: @liorn I would appreciate your opinion :)

**Pull request merged**